### PR TITLE
Add a tip on YAML anchors to docs on dataset layers

### LINKED ISSUE #2127: Document how to leverage YAML anchors & aliases to avoid copy-pasting properties in catalog
## Description
[This section](https://docs.kedro.org/projects/kedro-viz/en/stable/kedro-viz_visualisation.html#visualise-layers) of docs provides a guide to adding `layers` to the visualization by defining them as follows:

```yaml
companies:
  type: pandas.CSVDataset
  filepath: data/01_raw/companies.csv
  metadata:
    kedro-viz:
      layer: raw
```

Also it gives the following example below:

```yaml
companies:
  type: pandas.CSVDataset
  filepath: data/01_raw/companies.csv
  metadata:
    kedro-viz:
      layer: raw

reviews:
  type: pandas.CSVDataset
  filepath: data/01_raw/reviews.csv
  metadata:
    kedro-viz:
      layer: raw

shuttles:
  type: pandas.ExcelDataset
  filepath: data/01_raw/shuttles.xlsx
  metadata:
    kedro-viz:
      layer: raw

...
```

## Context
In my projects I found it very helpful to use YAML anchors to save those 3 lines per layer into a variable like this:

```yaml
_raw_layer: &raw_layer
  metadata:
    kedro-viz:
      layer: 01_raw
```

And then reuse it like this:

```yaml
companies:
  type: pandas.CSVDataset
  filepath: data/01_raw/companies.csv
  <<: *raw_layer

reviews:
  type: pandas.CSVDataset
  filepath: data/01_raw/reviews.csv
  <<: *raw_layer

shuttles:
  type: pandas.ExcelDataset
  filepath: data/01_raw/shuttles.xlsx
  <<: *raw_layer
```

## Possible Implementation
What I propose to do it to add a small **note** admonition suggesting that YAML anchors & aliases can be a great fit here to avoid copypasting those 3 lines if you have e.g. 10 datasets defined in a layer.

By admonition I mean e.g. this:

Screenshot 2024-06-24 at 10 35 00 PM

It can mention that this feature is not Kedro-specific at all and enabled by YAML format itself, but I think it can be helpful since this trick is highly reusable and can simplify large catalogs quite a lot for users unfamiliar with anchors & aliases in YAML.

> I do **not** propose to change the existing example which replicates those 3 lines 3 times.
> I think my suggestion better fits a note admonition.

## Checklist

- [x] Include labels so that we can categorise your feature request

### PR DESCRIPTION
**Pull request by yury-fedotov**
## Description

Closes #2127 

## Checklist

- [x] Test if docs render as expected
- [ ] Confirm that content matches expectations

### CODE REVIEW

File: docs/source/kedro-viz_visualisation.md [hunk 1]
```diff
@@ -194,6 +194,17 @@ The visualisation now includes the layers:
...
 
+```{tip}
+Duplicated defititions like
+
+metadata:
+  kedro-viz:
+    layer: raw
+
+can be avoided by leveraging YAML's native syntax for anchors and aliases.
+See [this example from the Kedro docs](https://docs.kedro.org/en/stable/data/data_catalog_yaml_examples.html#load-multiple-datasets-with-similar-configuration-using-yaml-anchors).
```
Comment (yury-fedotov) [inline docs/source/kedro-viz_visualisation.md:205]: Given that those are the docs of a separate project - I assume there's no way to link it other than through static URL?

**Comment by yury-fedotov** [comment]
Rendered docs: https://kedro--2181.org.readthedocs.build/projects/kedro-viz/en/2181/kedro-viz_visualisation.html#visualise-layers

image

We can't have `yaml` block nested inside RST `note` or `tip` block, so I decided to do just `yaml` instead.

**Comment by rashidakanchwala** [comment]
thanks @yury-fedotov  for this, will review this today, and fix the linting issues !

File: docs/source/kedro-viz_visualisation.md [hunk 2]
```diff
@@ -194,6 +194,17 @@ The visualisation now includes the layers:
 
 ![](./images/pipeline_visualisation_with_layers.png)
 
+Duplicated definitions like:
+
+```yaml
+metadata:
+  kedro-viz:
+    layer: raw
+```
```
Comment (rashidakanchwala) [inline docs/source/kedro-viz_visualisation.md:203]: can we also show the example here that you gave in the original issue on how to use it. thanks

Comment (rashidakanchwala) [reply docs/source/kedro-viz_visualisation.md:203]:
```yaml
_raw_layer: &raw_layer
  metadata:
    kedro-viz:
      layer: 01_raw
```

And then reuse it like this:

```yaml
companies:
  type: pandas.CSVDataset
  filepath: data/01_raw/companies.csv
  <<: *raw_layer

reviews:
  type: pandas.CSVDataset
  filepath: data/01_raw/reviews.csv
  <<: *raw_layer

Comment (yury-fedotov) [reply docs/source/kedro-viz_visualisation.md:203]: @rashidakanchwala yeah let me add this

**Comment by yury-fedotov** [comment]
Rendered docs: https://kedro--2181.org.readthedocs.build/projects/kedro-viz/en/2181/kedro-viz_visualisation.html#visualise-layers

**Review by ravi-kumar-pilla — APPROVED** [review]
Works well 💯 .. thank you @yury-fedotov

**Pull request merged**